app/src/main/java/org/covidwatch/android/presentation/HomeFragment.kt

```diff
@@ -95,6 +95,10 @@ class HomeFragment : Fragment(), EasyPermissions.PermissionCallbacks {
         binding.bannerText.setOnClickListener {
             homeViewModel.onBannerClicked()
         }
+        binding.swipeRefreshLayout.setOnRefreshListener {
+            homeViewModel.onRefreshRequested()
+            binding.swipeRefreshLayout.isRefreshing = false
```

Comment:
Not necessarily in this PR, but later we may want to show some message to Users that `contact_events` are updated

Reply:
We need a real feedback. This could lead to poor UX.

Let's use this: https://developer.android.com/topic/libraries/architecture/workmanager/how-to/states-and-observation

Reply:
@BurningAXE make `executePublicContactEventsRefresh` return `LiveData` 

```
fun executePublicContactEventsRefresh(): LiveData<Operation.State> {
    val downloadRequest = OneTimeWorkRequestBuilder<ContactEventsDownloadWorker>()
        .setConstraints(getRefreshConstraints())
        .build()

    val work = workManager.enqueueUniqueWork(
        "ContactEventsDownloadWorkerOneTimeRefresh",
        ExistingWorkPolicy.REPLACE,
        downloadRequest
    )
    return work.state
}
```

Then in ViewModel we can do something like this:
```
val refresh = Transformations.map(contactEventsDownloader.executePublicContactEventsRefresh()) {
    Event(Unit)
}
```

Reply:
Seeing your comment just now. Already added a commit that monitors the state of the work request. Let me know what you think.

Reply:
I'm trying to be objective and the solution that I proposed looks cleaner. 
Reasons:
- it uses builtin `state` that returns LiveData
- less code == less bugs

in `HomeFragment` 
you can just do this to make it work together with the code in my previous comment
```
binding.swipeRefreshLayout.setOnRefreshListener {
    homeViewModel.refresh.observe(viewLifecycleOwner, EventObserver{
        binding.swipeRefreshLayout.isRefreshing = false
    })
}
```

But I like the idea of having separate `LiveData` for refreshing status it could be handy if we have another sources of refresh. Although, for now is redundant.

Reply:
Hm, that looks strange to me. Why change status through an event? How do we set `isRefreshing = true` (if for example we do an automatic refresh at startup) with this implementation?

Reply:
Yes, that's why I was saying that the idea with `isRefreshing` is good to have. I missed the case when we would want to show refreshing status at startup but if talk about it let's do this then.

The main problem is that you should avoid passing references of a view to ViewModel. Which what you do with passing `viewLifecycleOwner` to the methods that goes into double dependency: view->viewmodel->ContactEventsDownloader

I propose add this MediatorLiveData livedata that will change the value of `isRefreshing` livedata:
```
private val _refresh = MediatorLiveData<Unit>().apply {
    addSource(contactEventsDownloader.executePublicContactEventsRefresh()) {
        _isRefreshing.value = false
    }
}

val refresh: LiveData<Unit>
    get() = _refresh
``` 
 
Then in `HomeFragment` 
```
binding.swipeRefreshLayout.setOnRefreshListener {
    homeViewModel.refresh.observe(viewLifecycleOwner, Observer {})
}
```

Reply:
It turned out that the Operation.State LiveData that is returned from enqueueing work is only for the operation of enqueueing. I.e. it is finished when the work is enqueued, not when the work is finished. So, I still used the `getWorkInfosForUniqueWorkLiveData` but passed it through two `MediatoLiveData`s to avoid passing the LifecycleOwner references. Looks too complex for a simple status monitoring but I think is architecturally sound now.

Reply:
Yes, with `MediatorLiveData` it's a bit complex but we can avoid it(see my comments).